.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,64 @@
+name: integration-tests
+on:
+  push:
+    branches: [ "518-implement-semi-automatic-integration-tests" ]
+  pull_request:
+    branches: [ "518-implement-semi-automatic-integration-tests" ]
+
+env:
+  STORAGE_PATH: "/tmp"
+
+jobs:
+  deploy-and-test:
+    strategy:
+      matrix:
+        key: [8, 16]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checks-out repo
+        uses: actions/checkout@v2
+      - name: Install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Install deps
+        run: |
+          /usr/bin/pip3 install -r ./requirements.txt
+        working-directory: ./integration-tests
+      - name: Install compose
+        uses: KengoTODA/actions-setup-docker-compose@main
+        with:
+          version: '2.6.0'
+      - name: Get version data
+        run: |
+          echo "VERSION=$(cat bob-apps/Cargo.toml | grep ^version | head -1 | awk '{print $3}' | sed -e 's/"//g' | sed -e 's/-/./g')" >> $GITHUB_ENV
+          echo "VERSION_NO_FORMATTING=$(cat bob-apps/Cargo.toml | grep ^version | head -1 | awk '{print $3}' | sed -e 's/"//g')" >> $GITHUB_ENV
+      - name: Get tools
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: build-packages.yml
+          workflow_conclusion: success
+          branch: v${{ env.VERSION_NO_FORMATTING }}
+          name: bob-binaries-${{ matrix.key }}
+          check_artifacts:  true
+      - name: Generate config files for docker and bob
+        env:
+          AMOUNT_OF_NODES: "4"
```

Comment:
You can use input parameters to allow customization by the user (https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs)